### PR TITLE
Deploy Opensearch

### PR DESCRIPTION
**Pull request by coilysiren**
## Summary

Fixes https://github.com/HHS/simpler-grants-gov/issues/1656

## Changes proposed

- Deploys an opensearch domain
- Configures the opensearch domain to be high availability in prod, and as cheap as possible (while still being similarly configured) elsewhere. Prod is currently turned off, though.
- Adds opensearch secrets to the API service

## Context for reviewers

Opensearch takes about 20 minutes to deploy, similarly to RDS Postgres

Generating the password was a pain in the butt

This style of this infra code is similar to the style of platform code, so I'm going to post it in our platform channel for other people to see

This PR tangentially includes the updated secrets handling from https://github.com/navapbc/template-infra/pull/602. I added that because I'm adding a bunch of secrets, and I don't want them to conflict with https://github.com/HHS/simpler-grants-gov/issues/2351.

## Additional information

Opensearch deployment:

(staging has since been torn down)

image

The terraform diff for the API service layer:

```
                    {
                        Action = "ssm:GetParameters"
                        Effect = "Allow"
                      ~ Resource = "arn:aws:ssm:*:*:parameter/api/dev/api-auth-token" -> [
                          + "arn:aws:ssm:us-east-1:315341936575:parameter/search/dev/username"
                          + "arn:aws:ssm:us-east-1:315341936575:parameter/search/dev/password"
                          + "arn:aws:ssm:us-east-1:315341936575:parameter/search/dev/endpoint"
                          + "arn:aws:ssm:us-east-1:315341936575:parameter/api/dev/api-auth-token"
                        ]
                        Sid = "SecretsAccess"
                    }
                ]
```

```
                    secrets = [
                        {
                        }
                      + {
                          + name = "SEARCH_ENDPOINT"
                          + valueFrom = "arn:aws:ssm:us-east-1:315341936575:parameter/search/dev/endpoint"
                        }
                      + {
                          + name = "SEARCH_PASSWORD"
                          + valueFrom = "arn:aws:ssm:us-east-1:315341936575:parameter/search/dev/password"
                        }
                      + {
                          + name = "SEARCH_USERNAME"
                          + valueFrom = "arn:aws:ssm:us-east-1:315341936575:parameter/search/dev/username"
                        }
                    ]
```

### CODE REVIEW

**Comment by coilysiren** [comment]
Related checkov Github issue: https://github.com/bridgecrewio/checkov/issues/6760

**Review by mdragon — DISMISSED** [review]
LGTM

File: infra/api/service/main.tf [hunk 1]
```diff
@@ -138,8 +138,26 @@ module "service" {
     }
   } : null
 
-  extra_environment_variables = merge(local.service_config.extra_environment_variables, { "ENVIRONMENT" : var.environment_name })
```
Comment (chouinar) [inline infra/api/service/main.tf:141]: It looks like you undid the change you made here

File: infra/api/service/main.tf [hunk 2]
```diff
@@ -138,8 +138,26 @@ module "service" {
...
+    }] : [],
+    local.environment_config.search_config != null ? [{
+      name      = "SEARCH_PASSWORD"
+      valueFrom = data.aws_ssm_parameter.search_password_arn[0].arn
+    }] : [],
+    local.environment_config.search_config != null ? [{
+      name      = "SEARCH_ENDPOINT"
+      valueFrom = data.aws_ssm_parameter.search_endpoint_arn[0].arn
+    }] : []
+  )
```
Comment (chouinar) [inline infra/api/service/main.tf:160]: What port do we connect to OpenSearch on?

**Review by chouinar — COMMENTED** [review]
Just two small env var related comments

File: infra/api/service/main.tf (hunk 1 above)
Comment (coilysiren) [reply infra/api/service/main.tf:141]: Ahah! Thanks

File: infra/api/service/main.tf (hunk 2 above)
Comment (coilysiren) [reply infra/api/service/main.tf:160]: It didn't tell me! Whatever the default is

Comment (coilysiren) [reply infra/api/service/main.tf:160]: @chouinar env vars thing fixed o7

**Review by chouinar — APPROVED** [review]

**Pull request merged**